qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -246,7 +251,16 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
                 )
 
             all_circuit_results.append(circuit_results)
-            oracle_evaluation = amplification_problem.is_good_state(top_measurement)
+
+            # only check if top measurement is a good state if an _is_good_state arg is provided
+            if hasattr(amplification_problem, "_is_good_state"):
```

Comment:
We shouldn't really use private-member access of a different class - it's usually fragile, and it fuzzes the API boundaries between classes.  If you want to do it here, it's likely that the user may want to do it as well.  It might be better to have the public `AmplificationProblem.is_good_state` be `None` if it's undecidable, and test against that value.